subprojects/language-jvm/src/testFixtures/groovy/org/gradle/api/tasks/compile/AbstractCompilerDaemonReuseIntegrationTest.groovy

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.compile
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.jvm.TestJvmComponent
+import org.gradle.test.fixtures.file.TestFile
+
+
+abstract class AbstractCompilerDaemonReuseIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
Yeah, I added this because I couldn't find any test coverage that we were actually reusing compiler daemons within a build and I actually got through several stages of CI before I realized that I had a bug that was preventing reuse in some cases.

We can make this coverage better, of course.